lib/synapse/service_watcher/zookeeper.rb

```diff
@@ -127,7 +127,12 @@ def discover
 
       new_backends = []
       @zk.children(@discovery['path'], :watch => true).each do |id|
-        node = @zk.get("#{@discovery['path']}/#{id}")
+        begin
+          node = @zk.get("#{@discovery['path']}/#{id}")
+        rescue ZK::Exceptions::NoNode => e
+          log.error("synapse: consistency issue reading #{id} at #{@discovery['path']}: #{e}")
```

Comment:
something like `"synapse: node #{id} disappeared before it could be read"` is probably fine, along with a comment in the code explaining why this could happen.